src/shared/seccomp-util.c

```diff
@@ -699,6 +700,50 @@ static int seccomp_add_syscall_filter_set(
         return 0;
 }
 
+int seccomp_add_syscall_filter_set_whitelist(
+                scmp_filter_ctx seccomp,
+                uint32_t default_action,
+                const SyscallFilterSet *set,
+                char **syscall_whitelist,
+                uint32_t action) {
+
+        const char *sys;
+        int r;
+
+        assert(seccomp);
+        assert(set);
+
+        NULSTR_FOREACH(sys, set->value) {
+                int id;
+
+                /* We can also whitelist entire filter sets */
+                if (sys[0] == '@' && !strv_contains(syscall_whitelist, sys)) {
+                        const SyscallFilterSet *other;
+
+                        other = syscall_filter_set_find(sys);
+                        if (!other)
+                                return -EINVAL;
+
+                        r = seccomp_add_syscall_filter_set_whitelist(seccomp, default_action, other, syscall_whitelist, action);
+                        if (r < 0)
+                                return r;
+                } else {
+                        if (!strv_contains(syscall_whitelist, sys)) {
+                                id = seccomp_syscall_resolve_name(sys);
+                                if (id == __NR_SCMP_ERROR)
+                                        return -EINVAL; /* Not known at all? Then that's a real error */
+
+                                r = seccomp_rule_add_exact(seccomp, action, id, 0);
+                                if (r < 0)
+                                        /* If the system call is not known on this architecture, then that's fine, let's ignore it */
+                                        log_debug_errno(r, "Failed to add rule for system call %s, ignoring: %m", sys);
+                        }
+                }
+        }
+
+        return 0;
+}
+
```

Comment:
why is this function separate of seccomp_add_syscall_filter_set()? It appears you only need one, and the other can be replaced by a tiny stub in the header that calls the other with a NULL whitelist, no?